active/InputValidation/InputValidation.md

```diff
@@ -0,0 +1,592 @@
+
+## InputValidationMode
+Determines how the ValidationCommand should perform validation.
+
+    /// <summary>
+    ///    Determines how the ValidationCommand should perform validation
+    /// </summary>
+    [webhosthidden]
+    enum InputValidationMode
+    {
+        /// <summary>
+        /// Only validate on PropertyChanged once there are errors
+        /// </summary>
+        Default,
+        /// <summary>
+        /// Validation is disabled.
+        /// </summary>
+        Disabled,
+    };
+
+
+## IInputValidationControl
+Any control that wants to participate in input validation should derive from this interface. This, among other things, is what allows the markup compiler to know when to generate the code necessary for listening to the `INotifyDataErrorInfo.ErrorsChanged` event and propagate the results to the `IInputValidationControl.Errors` property.
+
+When the `ValidationCommand` property is non-null, the following will be true:
+1. The control tries to invoke the command whenever its input property changes.
+1. The control uses the return value of `ValidationCommand.CanValidate` to determine if it can invoke the `ValidationCommand.Validate` method
+2. The default implementation of `ValidationCommand.CanValidate` returns false when the `InputValidationMode` property is set to `Disabled`, and only returns true if the he `IInputValidationControl.ValidationErrors` collection is non-empty. Custom commands are allowed to override this default behavior as they see fit.
+3. When the `InputValidationKind` property is set to `Auto`, the control will use the value of the `ValidationCommand.InputValidationKind` property for the value. If both are `Auto` the default bevahior should be `Inline`.
+
+The markup compiler uses this interface to properly generate code when using x:Bind if and only if:
+1. The Source implements INotifyDataErrorInfo and INotifyPropertyChanged 
+2. The Target property is the one specified by the TargetType's InputPropertyAttribute
+3. The BindingMode is set to TwoWay
+
+The code generated provides the following functions:
+1. Generate code that listens to the Source's INotifyDataErrorInfo.ErrorsChanged event and populates the Target's 'IInputValidationControl.ValidationErrors' property.
+2. Generate and automatically set the ValidationContext property based on the Source property. If that property has any System.ComponentModel.DataAnnotations.ValidationAttributes, the associated property on the ValidationContext class will be set.
+
+         /// <summary>
+         /// Interface for Controls that participate in input validation.
+         /// <summary/>
+          [webhosthidden]
+         interface IInputValidationControl 
+         {
+             /// <summary>
+             /// Collection of errors to display based on the InputValidationKind property.
+            /// </summary>
+            Windows.Foundation.Collections.IObservableVector<Windows.UI.Xaml.Controls.InputValidationError> ValidationErrors{ get; };
```

Comment:
Could you update this comment to describe where the errors come from? Does the control itself populate them? Does the XAML compiler generated code set them?

Reply:
they are populated by the app - either via the compiler if using x:Bind, or the app author